src/common/regions.ts

```diff
@@ -0,0 +1,146 @@
+import { chain, fromPairs } from 'lodash';
+import urlJoin from 'url-join';
+import US_STATE_DATASET from 'components/MapSelectors/datasets/us_states_dataset_01_02_2020.json';
+import countyAdjacencyMsa from './data/county_adjacency_msa.json';
+const { state_dataset, state_county_map_dataset } = US_STATE_DATASET;
+
+export type FipsCode = string;
+
+export enum RegionType {
+  COUNTY = 'County',
+  STATE = 'State',
+  MSA = 'MSA',
+}
+
+abstract class Region {
+  constructor(
+    public name: string,
+    public urlName: string,
+    public fipsCode: FipsCode,
+    public population: number,
+    public regionType: RegionType,
+  ) {}
+
+  abstract fullName(): string;
+  abstract relativeUrl(): string;
+  canonicalUrl() {
+    return urlJoin('https://covidactnow.org', this.relativeUrl());
+  }
+}
+
+export class State extends Region {
+  constructor(
+    public name: string,
+    public urlName: string,
+    public fipsCode: FipsCode,
+    public population: number,
+    public stateCode: string,
+  ) {
+    super(name, urlName, fipsCode, population, RegionType.STATE);
+  }
+
+  fullName() {
+    return this.name;
+  }
+
+  relativeUrl() {
+    return `us/${this.urlName}`;
+  }
+}
+
+export class County extends Region {
+  constructor(
+    public name: string,
+    public urlName: string,
+    public fipsCode: FipsCode,
+    public population: number,
+    public state: State,
+    public cities: string[],
```

Comment:
small, but `cityNames` might be more clear here.  Based on `state` above being an object, my first thought was that `cities` might also be Region objects

Reply:
Makes sense, will update!